components/list/list-item-drag-drop-mixin.js

```diff
@@ -47,8 +47,10 @@ export const ListItemDragDropMixin = superclass => class extends superclass {
 			draggable: { type: Boolean, reflect: true },
 			dragging: { type: Boolean, reflect: true },
 			dragHandleText: { type: String, attribute: 'drag-handle-text' },
+			dropDestination: { type: Number, reflect: true },
 			dropText: { type: String, attribute: 'drop-text' },
 			key: { type: String, reflect: true },
+			size: { type: Number, reflect: true },
```

Comment:
I'm not really a fan of adding the size to the item component since it results in two sources of truth.  This is really a property of the list, which ought to be obtained by querying the list item count when needed.